Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,463 @@
+                    }
+                }
+                _npcFaction.RemoveFaction(mind.OwnedEntity.Value, "NanoTrasen", false);
+                _npcFaction.AddFaction(mind.OwnedEntity.Value, "Revolutionary");
+                inCharacterName = MetaData(mind.OwnedEntity.Value).EntityName;
+                comp.HeadRevs.Add(inCharacterName, headRev.Name);
+                _audioSystem.PlayGlobal(comp.HeadRevStartSound, mind.OwnedEntity.Value);
+            }
+            if (mind.Session != null)
+            {
+                var message = Loc.GetString("head-rev-role-greeting");
+                var wrappedMessage = Loc.GetString("chat-manager-server-wrap-message", ("message", message));
+                _chatManager.ChatMessageToOne(Shared.Chat.ChatChannel.Server, message, wrappedMessage, default, false, mind.Session.ConnectedClient, Color.FromHex("#5e9cff"));
+            }
+        }
+    }
+
+    /// <summary>
+    /// Assigns command staff on station with a component so they can be used later.
+    /// </summary>
+    private void AssignCommandStaff()
+    {
+        var jobs = _prototypeManager.Index<DepartmentPrototype>("Command");
+        var allPlayers = _playerSystem.ServerSessions.ToList();
+        var playerList = new List<IPlayerSession>();
+        foreach (var player in allPlayers)
+        {
+            if (player.AttachedEntity == null || HasComp<HumanoidAppearanceComponent>(player.AttachedEntity))
+            {
+                playerList.Add(player);
+            }
+        }
+        foreach (var player in playerList)
+        {
+            var mind = player.GetMind();
+            if (mind != null && mind.CurrentJob != null)
+            {
+                var currentJob = mind.CurrentJob.Name;
+                currentJob = currentJob.Replace(" ", "");
+                if (mind.OwnedEntity != null && jobs.Roles.Contains(currentJob))
+                {
+                    if (!HasComp<HeadRevolutionaryComponent>(mind.OwnedEntity))
+                    {
+                        EnsureComp<RevolutionaryRuleComponent>(mind.OwnedEntity.Value);
+                        EnsureComp<CommandStaffComponent>(mind.OwnedEntity.Value);
+                    }
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Called when a Head Rev uses a flash in melee to convert somebody else.
+    /// </summary>
+    public void OnPostFlash(EntityUid uid, HeadRevolutionaryComponent comp, ref AfterFlashedEvent ev)
+    {
+        var stunTime = TimeSpan.FromSeconds(3);
+        if (!HasComp<RevolutionaryComponent>(ev.Target) && !HasComp<MindShieldComponent>(ev.Target) && HasComp<HumanoidAppearanceComponent>(ev.Target)
+            && TryComp<MobStateComponent>(ev.Target, out var mobState) && mobState.CurrentState == MobState.Alive)
+        {
+            if (ev.Used != null)
+            {
+                _charges.AddCharges(ev.Used.Value, 1);
+            }
```

Comment:
Isn't this going to give them infinite charges? They should be getting their own flash.

Reply:
It only regains the charge if it does a successful conversion but if you really think I should make another flash I guess I can.